File: src/scala/NumberReverse.scala
object NumberReverse {
  def reverse(x: Int): Int = {
    var result = 0
    var reverseNumStr = ""
    if (x == 0) return x
    else if (x < 0) {
      val sb = new StringBuilder(-1 * x + "")
      reverseNumStr = "-" + sb.reverse.toString
    }
    else {
      val sb = new StringBuilder(x + "")
      reverseNumStr = sb.reverse.toString
    }
    try
      result = reverseNumStr.toInt
    catch {
      case e: NumberFormatException =>
    }
    result
  }
}

File: src/java/NumberReverse.java
/**
 * 7、数字反转
 * @author LJY
 */
public class NumberReverse {

    public int reverse(int x) {
        int result = 0;
        String reverseNumStr;
        if(x == 0){
            return x;
        }else if(x < 0){
            StringBuilder sb = new StringBuilder(-1 * x + "");
            reverseNumStr = "-" + sb.reverse().toString();
        }else{
            StringBuilder sb = new StringBuilder(x + "");
            reverseNumStr = sb.reverse().toString();
        }
        try {
            result = Integer.parseInt(reverseNumStr);
        } catch (NumberFormatException e) {
        }
        return result;
    }

}
